src/main/scala/org/constellation/util/Metrics.scala

```diff
@@ -178,6 +178,12 @@ class Metrics(periodSeconds: Int = 1)(implicit dao: DAO) extends Periodic[Unit](
   def updateMetricAsync[F[_]: Sync](key: String, value: Long): F[Unit] = Sync[F].delay(updateMetric(key, value))
   def incrementMetricAsync[F[_]: Sync](key: String): F[Unit] = Sync[F].delay(incrementMetric(key))
 
+  def incrementMetricAsync[F[_]: Sync](key: String, either: Either[Any, Any]): F[Unit] =
```

Comment:
Can you make it similar to those fns which currently exist? `tryWithMetric`

Reply:
Reason why it's not this way is due to retry logic/removal of snapshots so the metric update goes after when the final result is being calculated.